packages/artillery/test/cli/command-run.test.js

```diff
@@ -0,0 +1,207 @@
+  t.ok(
+    output.stdout.includes('Summary report')
+  );
+});
+
+tap.test('Run a script with one payload json config with parse options passed', async (t) => {
+  const [exitCode, output] = await execute(['run', 'test/scripts/single_payload_options.json']);
+
+  t.ok(
+    exitCode === 0
+    && output.stdout.includes('Summary report')
+  );
+});
+
+tap.test('Run a script with multiple payloads and use of $environment in path', async (t) => {
+  const [exitCode, output] = await execute(['run', '--environment', 'local', 'test/scripts/multiple_payloads.json']);
+
+  t.ok(
+    exitCode === 0
+    && output.stdout.includes('Summary report')
+  );
+});
+
+tap.test('Run a script overwriting default options (output)', async (t) => {
+  const reportFile = 'artillery_report_custom.json';
+  const [exitCode, output] = await execute(['run', '--config', 'test/scripts/hello_config.json', 'test/scripts/hello.json', '-o', reportFile]);
+
+  t.ok(
+    exitCode === 0
+    && output.stdout.includes('Log file: artillery_report_custom.json')
+  );
+});
+
+tap.test('Script using hook functions', async (t) => {
+  const [exitCode, output] = await execute(['run', '--config', 'test/scripts/hello_config.json', 'test/scripts/hello.json']);
+
+  t.ok(
+    exitCode === 0
+    && output.stdout.includes('hello from processor')
+  );
+});
+
+tap.test('Hook functions - can rewrite the URL', async (t) => {
+  // Ref: https://github.com/shoreditch-ops/artillery/issues/185
+  const reportFile = 'report-hook.json';
+  const reportFilePath = await getRootPath(reportFile);
+  const [exitCode, output] = await execute(['run', '--config', 'test/scripts/hello_config.json', 'test/scripts/hello.json', '-o', reportFile]);
+  const json = JSON.parse(fs.readFileSync(reportFilePath, 'utf8'));
+
+  t.ok(
+    deleteFile(reportFilePath)
+    && exitCode === 0
+    && json.aggregate.counters['http.codes.200']
+  );
+});
+
+tap.test('Environment variables can be loaded from dotenv files', async (t) => {
+  const reportFile = 'report-with-dotenv.json';
+  const reportFilePath = await getRootPath(reportFile);
+  const [exitCode, output] = await execute(['run', '--dotenv', 'test/scripts/with-dotenv/my-vars', 'test/scripts/with-dotenv/with-dotenv.yml', '-o', reportFile]);
+  const json = JSON.parse(fs.readFileSync(reportFilePath, 'utf8'));
+
+  t.ok(
+    deleteFile(reportFilePath)
+    && exitCode === 0
+    && json.aggregate.counters['http.codes.200']
+  );
+});
+
+tap.test('Script using a plugin', async (t) => {
+  const reportFile = 'report-with-plugin.json';
+  const reportFilePath = await getRootPath(reportFile);
+  const pluginFile = 'plugin-data.csv';
+  const pluginFilePath = await getRootPath(pluginFile);
+
+  const pluginPath = path.resolve(
+    __dirname,
+    '..',
+    'plugins'
+  );
+
+  const abortController = new AbortController();
+  const subprocess = execa('node', ['../core/test/core/targets/simple.js'], {env:{'PORT': '3003'}, signal: abortController.signal});
```

Comment:
this allows us to run single tests if we want to.
But can be mover to runner.sh if we prefer having them available to the whole suite